rotkehlchen/data_import/utils.py

```diff
@@ -75,9 +118,44 @@ def flush_all(self, write_cursor: DBCursor) -> None:
         self._asset_movements = []
         self._history_events = []
 
+    def append_msg(self, row_index: int, msg: str, warning: bool) -> None:
+        """Append message to queue to be sent to frontend"""
+        if self.max_msgs:
+            return
+
+        # Stop collecting msgs if more than MAX_ERROR_PERCENT have errors.
+        # Note that total_entries actually only contains the number of rows processed do far.
+        # So if the percentage of errors is too high in the first MIN_ENTRIES rows
+        # error collecting will stop even if the rest of the file has few errors.
+        if (
+            self.total_entries >= MIN_ENTRIES and
+            len(self.import_msgs) >= self.total_entries * MAX_ERROR_PERCENT
+        ):
+            self.max_msgs = True
+
+        self.import_msgs.append({
+            'row': row_index,
+            'msg': msg,
+            'warning': warning,
+        })
+
+    def log_debug(self, row_index: int, csv_row: dict, msg: str) -> None:
```

Comment:
Why do you need log debug? Do we have both errors and warnings? Shouldn't we then have different functions?

Reply:
Its only supposed to be using debug and warning, not error. The basic idea is that warnings are for rows that we couldn't import but may have had valuable data in them, while debug is for rows that we skip importing because we can tell they don't have any data worth importing (for example, [here](https://github.com/rotki/rotki/blob/21ca43e13af1e84248ab3f32f4e74ee32abe6c50/rotkehlchen/data_import/importers/cryptocom.py#L284) it is logged via debug that an entry has been skipped because is doesn't affect the wallet balance).

As was mentioned [here](https://github.com/rotki/rotki/pull/8458#discussion_r1736971529), I used `log.debug` and `log.warning` since that is what was being used in the importers already, also it would seem to me that if something is logged with `log.error` that should mean that the entire csv import operation failed instead of just a single entry failing to import.